applications/bluesky.md

```diff
@@ -0,0 +1,226 @@
+| 3. | Substrate modules | implement substrate custom pallets corresponding to above macros |
+| 4. | Smart contracts | Write Ink! smart contract with macros integrated with custom pallets to fulfil use case business functions |
+| 5. | End-to-end integration | Organically integrate all pieces together to ensure working end-to-end |
+| 6. | Deployment | Deploy to a Polkadot smart contract parachain, say Shiden, etc to also explore interoperability |
+| 7. | Articles | We will publish article 1 of 3 walking through the technical showcase focusing on Ink! macros and substrate pallets. Can publish the blog in Medium in English, or recommended medium to target designated audiences |
+
+### Milestone 2 —  On-chain Ink! smart contract upgrades, compared to Solidity upgrades. Publish Article 2/3   
+
+- **Estimated Duration:** 2 month
+- **FTE:**  2,3
+- **Costs:** 50,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 1. | Upgrade Ink! smart contracts due to feature changes | Demonstrate Upgradability via ink!'s built-in upgradeable smart contract features to upgrade the smart contract code without having to migrate the data or interrupt the user experience |
+| 2. | Seamless interaction during upgrades | Demonstrate users can continue to interact with the updated contract without losing any data or transaction history as Ink!'s flexible upgrade mechanism allows for on-chain upgrades of smart contracts |
+| 3. | Demo Ink! built-in tools and libraries for managing contract upgrades | For example, the "Upgradeable" trait that can be used to mark a smart contract as upgradeable, and the "Dispatcher" module that can be used to handle contract upgrades and routing of function calls|
+| 4. | Expose Ink!'s Dispatcher module | To handle contract upgrades and routing of function calls between different versions of the same contract. When a contract is upgraded, the Dispatcher module creates a new contract instance on the blockchain for the upgraded version of the contract |
+| 5. | Copying data from the old contract | Migrate data from the old contract instance to the new one, demonstrating copying data from the old contract instance to the new one, using migrate function|
+| 6. | Demo Ink! upgradability and parachain flexibility without hard fork | Compare to Solidity smart contract upgrades |
+| 7. | Articles | We will publish article 2 of 3 using the same technical showcase focusing on Ink! smart contract upgrade flow, ease of upgradability and flexibility. Can publish the blog in Medium in English, or recommended medium to target designated audiences |
+
+
+### Milestone 3 —  Gas usage / performance benchmarks compared, Ink!+WASM vs Solidity+EVM, Publish Article 3/3
+
+- **Estimated Duration:** 2 month
+- **FTE:**  3+
+- **Costs:** 60,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0 / GPLv3 / MIT / Unlicense |
```

Comment:
What will be unlicensed? What will be open source?

Reply:
All open source, nothing unlicensed.